src/grubconfigs.rs

```diff
@@ -103,6 +106,7 @@ pub(crate) fn install(
             efidir
                 .copy_file(&Path::new(CONFIGDIR).join("grub-static-efi.cfg"), target)
                 .context("Copying static EFI")?;
+            efidir.set_mode(target, GRUBCONFIG_FILE_MODE)?;
```

Comment:
Side note: in the future we could use https://docs.rs/cap-std-ext/latest/cap_std_ext/dirext/trait.CapStdExtDirExt.html#tymethod.atomic_write_with_perms to do this in one step.